src/consumption/elevate.js

```diff
@@ -0,0 +1,1827 @@
+		// Mark socket as in use
+		socketInUse = true
+
+		// Download the image file as a binary buffer
+		const response = await axios({
+			url,
+			method: 'GET',
+			responseType: 'arraybuffer', // Ensures we receive raw binary data
+			timeout: 120000,
+			headers: {
+				Connection: 'close', // Ensures the socket is closed after the request
+			},
+		})
+
+		// Convert the binary data to a Base64 string
+		const base64 = Buffer.from(response.data, 'binary').toString('base64')
+
+		// Create the Base64 Data URL
+		const base64DataUrl = `data:image/png;base64,${base64}`
+
+		// Mark socket as free
+		socketInUse = false
+
+		return base64DataUrl
+	} catch (error) {
+		console.error('Error downloading or converting file:', error.message)
+		// Mark socket as free
+		socketInUse = false
+		throw error
+	}
+}
+
+async function generatePresignedUrlInConsumption(url, body, headers) {
+	try {
+		const response = await axios.post(url, body, { headers, timeout: 6000 })
+		let result = { success: false }
+		if (response.status === 200) {
+			const files = response?.data?.result?.[common.CERTIFICATE]?.files
+
+			if (Array.isArray(files) && files.length > 0) {
+				result.file = files[0]?.payload?.sourcePath || null
+				result.url = files[0]?.url || null
+				result.success = true
+			} else {
+				console.error('Files array is missing or empty:', files)
+			}
+		} else {
+			console.error('Unexpected response status:', response.status)
+		}
+
+		return result
+	} catch (error) {
+		console.error('Error generating consumption presigned URL:', error.message)
+		throw error // Rethrow the error to be handled by the caller
+	}
+}
+
+/**
+ * create svg template by editing base template.
+ * @method
+ * @name createSvg
+ * @param {Object} certificateData - Certificate data for upload
+ */
+
+async function createSvg(certificateData, loggedInUserId, userToken) {
+	return new Promise(async (resolve, reject) => {
+		try {
+			// fetch base template from cloud
+			let baseTemplate = await getBaseTemplate(certificateData?.base_template_url)
+			if (!baseTemplate.success) {
+				throw new Error('Base template download failed.')
+			}
+
+			// Load SVG template using Cheerio with XML mode
+			const $ = cheerio.load(baseTemplate.result, { xmlMode: true })
+
+			// set issuer name
+			const issuerNameTag = 'stateTitle'
+			const issuerNameElement = $(`#${issuerNameTag}`)
+			issuerNameElement.text(utils.escapeXml(certificateData.issuer))
+
+			// update signature
+			for (let index = 1; index <= certificateData.signature.no_of_signature; index++) {
+				const signatureNameTag = `signatureTitle${index}a`
+				const signatureDesignationTag = `signatureTitleDesignation${index}`
+				const signatureImgTag = `signatureImg${index}`
+				await waitForSocketAvailability() // check and wait for axios socket availability
+				const imageData = await downloadAndConvertToBase64(certificateData.signature[signatureImgTag])
+				const signatureNameElement = $(`#${signatureNameTag}`)
```

Comment:
@adithyadinesh0412 whatever new keys which Ram asked in between is added here as well right ?